docs/Operators.md

```diff
@@ -7611,17 +7611,213 @@ Other versions of this operator: <a href="Changelog.md#GreaterOrEqual-12">12</a>
+dimension, `-1` refers to the center of the left-most column of pixels while `1`
+refers to the center of the right-most column of pixels. The formula for
+denormalization of a coordinate `x` is then:
+
+  `denormalize(x, W, true) = (x + 1) * (W - 1) / 2` .
+
+The normalized coordinates in `grid` are notated as `(x, y)` while the
+denormalized coordinates used for lookups are notated as `(row, col)`, matching
+PyTorch's GridSample. The concept of `x` is vertical (left-right) which
+corresponds to `col`, while `y` is horizontal (up-down) which corresponds to
+`row`. So in order to do a lookup in `input` based on `(x, y)`, it is also
+necessary to swap the two coordinates in addition to denormalization. All
+together, the coordinate calculation is then:
+
+  `(row, col) = (denormalize(y, H, align_corners), denormalize(x, W, align_corners))`
+
+where `y` goes first while in `grid` the ordering is `(x, y)` with `x` first. In
+other words, `grid` lists its coordinates in minor-to-major ordering (`y` and
+`row` are the major dimensions while `x` and `col` are minor). This is in
+contrast to e.g. `gather_nd` which uses the opposite ordering. So if `(-1, -1)`
+is the top-left corner, then `(-1, 1)` refers to the bottom-left corner, not the
+top-right corner.
+
+#### Interpolation of coordinates depending on `interpolation_mode`
+
+If the denormalized coordinates `(row, col)` are integer and in-bounds, then the
+value in the output at those coordinates is the value of `input` at those
+coordinates. If the denormalized coordinates are not integer, then the value is
+interpolated depending on the value of `interpolation_mode`.
+
+If `interpolation_mode` is `nearest`, then the value at those
+coordinates are the values in `input` of the nearest integer coordinates. This
+corresponds to the calculation
+
+`(row', col') = (round(row), round(col))` .
+
+If two integer points are equally close in a given dimension,
+corresponding to a coordinate ending in `.5`, then a rounding occurs
+according to the value of the nearest_mode attribute.
+
+For example, if `(row, col)=(1.5,3.5)`, then the coordinates used for the lookup
+could be any one of `(1.0, 3.0)`, `(1.0, 4.0)`, `(2.0, 3.0)`, and `(2.0, 4.0)`.
+Whichever one of those a given version of an implementation picks, it must
+always make that same choice when given this same input.
+
+If `interpolation_mode` is `bilinear`, then the value at the output depends on a
+linear interpolation between the nearest four integer coordinates. The weight of
+the nearby integer points depends on the (Manhattan) distance to those points.
+In the horizontal (left-right) dimension, the weighting of the two nearest
+points to the left equals `frac(col)` which is the fractional part of `col`
+while the weighting of the two nearest points to the right equals `1 -
+frac(col)`.
+
+For example, if `(row, col) = (1.3, 3.9)` then the interpolated value is the
+matrix product:
+
+```math
+\text{output[n,c,row',col'] } =
```

Comment:
Not Sure if there are someway to display the `mathematical formula` well in markdown file.

Reply:
On gitlab this example (or something like it, I tried a few different versions before putting this up here) displays as beautiful LaTeX rendered math. Seems like we'd want to be able to describe math in something like a ML op spec since everything is very mathematical in ML?

There are some people with thoughts on that here, there appear to be some workarounds possible on github:

  https://stackoverflow.com/questions/35498525/latex-rendering-in-readme-md-on-github

Maybe we could have an HTML mode or something where this works?

In any case, we should have an answer about how to describe math if we don't have typesetting for it. Maybe some sort of plain ascii-based math?

Reply:
Hi @broune, 
After my confirmation, ONNX spec file operator.md does not support the LaTeX formula so for. But there are some plans to support it in the next half year, and some of the limitations of the operators will be fixed at the same time.
Just as you said, we might have to describe math like the way of [RNN-14](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#rnn-14) describing the math of activation functions.

Reply:
This now works.